### DISCUSSION TITLE
mihomo 连接中，无域名，全部匹配到IP

### DISCUSSION THREAD

**Discussion by xixi-furry** [MetaCubeX/mihomo#1671] [Q&A]
用的是mihomoTproxy，用的是tproxy代理，dns由adguard负责，没让clash来接手。之前的openclash和现在的mihomoTproxy都是一样的

**Comment by lonble** [comment]
如果使用tproxy，或者tun，或者使用socks但没选代理DNS，那么mihomo只能看到IP地址。如果想用域名规则，要么配置http或者socks5h代理，要么开启嗅探或者fakeip

**Reply by xixi-furry** [reply]
开启嗅探后确实解决了，但是这个fakeip。似乎还是得接管dns才能处理吧x

**Reply by lonble** [reply]
fakeip需要接管dns，而且会导致dig等DNS诊断工具没法用。
嗅探只能处理http、tls、quic这种请求里带域名的，ssh这类的处理不了

**Reply by xixi-furry** [reply]
> fakeip需要接管dns，而且会导致dig等DNS诊断工具没法用。 嗅探只能处理http、tls、quic这种请求里带域名的，ssh这类的处理不了

谢谢大佬，主要我工作需求有dig或者ping的这种需求，那我还是考虑去掉adg吧哈哈哈哈哈，主要是想做广告拦截，但是如果所有流量都走一遍mihomo然后再让op丢到ikuai去，这两个vm一起跑软路由cpu基本就满了，如果大流量的情况下。

**Reply by Plutonium141** [reply]
> 谢谢大佬，主要我工作需求有dig或者ping的这种需求，那我还是考虑去掉adg吧哈哈哈哈哈，主要是想做广告拦截，但是如果所有流量都走一遍mihomo然后再让op丢到ikuai去，这两个vm一起跑软路由cpu基本就满了，如果大流量的情况下。

利用mihomo的规则和dns也可以拦截广告的

**Reply by xixi-furry** [reply]
> > 谢谢大佬，主要我工作需求有dig或者ping的这种需求，那我还是考虑去掉adg吧哈哈哈哈哈，主要是想做广告拦截，但是如果所有流量都走一遍mihomo然后再让op丢到ikuai去，这两个vm一起跑软路由cpu基本就满了，如果大流量的情况下。
> 
> 利用mihomo的规则和dns也可以拦截广告的

感谢大佬的指导，我之前是让规则去负责的，但是一旦国内域名也走openclash，流量一大就会让cpu满载，ikuai和openwrt两个vm满载，就会卡。主要是这个情况
